app/Domain/Auth/Actions/LoginUserAction.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domain\Auth\Actions;
+
+use App\Exceptions\UserNotFoundException;
+use App\Models\User;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Validation\ValidationException;
+
+class LoginUserAction
+{
+    public function __construct(protected User $user)
```

Comment:
$user property is not used?

Reply:
this is used in the execute method im just injected the model here to use the container instead of `User::` in the code inside of the execute method.